src/app/clusters/mode-base-server/mode-base-server.h

```diff
@@ -229,9 +239,10 @@ class Delegate
 
 // A set of pointers to all initialised ModeBase instances. It provides a way to access all ModeBase derived clusters.
 // todo change once there is a clear public interface for the OnOff cluster data dependencies (#27508)
-static std::set<Instance *> ModeBaseAliasesInstances;
+static IntrusiveList<Instance> gModeBaseAliasesInstances;
 
-std::set<Instance *> * GetModeBaseInstances();
+// This does not return a const of gModeBaseAliasesInstances since the caller might need to call the element's UpdateXxx methods.
```

Comment:
```suggestion
// This does not return a refernce to const IntrusiveList, because the caller might need
// to change the state of the instances in the list and const IntrusiveList only allows
// access to const Instance.
```